### PR TITLE
feat(instance): add support for bootscript

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: scaleway/resource_instance_server.go [hunk 1]
```diff
@@ -174,9 +174,14 @@ func resourceScalewayInstanceServer() *schema.Resource {
 			},
 			"boot_type": {
 				Type:        schema.TypeString,
-				Computed:    true,
+				Optional:    true,
 				Description: "The boot type of the server",
 			},
+			"bootscript": {
```
Comment (kindermoumoute) [inline scaleway/resource_instance_server.go:180]: bootscript_id

File: scaleway/resource_instance_server.go [hunk 2]
```diff
@@ -174,9 +174,14 @@ func resourceScalewayInstanceServer() *schema.Resource {
 			},
 			"boot_type": {
 				Type:        schema.TypeString,
-				Computed:    true,
+				Optional:    true,
 				Description: "The boot type of the server",
```
Comment (kindermoumoute) [inline scaleway/resource_instance_server.go:178]: add default value

File: scaleway/resource_instance_server.go [hunk 3]
```diff
@@ -259,6 +264,12 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 		Tags:              expandStrings(d.Get("tags")),
 	}
 
+	if bootType, ok := d.GetOk("boot_type"); ok && bootType == "bootscript" {
```
Comment (kindermoumoute) [inline scaleway/resource_instance_server.go:267]: [use bootType const](https://github.com/scaleway/scaleway-sdk-go/blob/3c87d958dd1172db3151d8fdbbb51eb309b1bd32/api/instance/v1/instance_sdk.go#L92)

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: scaleway/resource_instance_server.go (hunk 2 above)
Comment (Sh4d1) [reply scaleway/resource_instance_server.go:178]: maybe add the allowed values too ?

**Comment by Sh4d1** [comment]
What happens when a user specifies a bootscript without changing the boot type ? We could either throw an error, or change the boot type.

File: docs/resources/instance_server.md [hunk 4]
```diff
@@ -177,7 +177,7 @@ attached to the server. Updates to this field will trigger a stop/start of the s
 
 - `boot_type` - The boot Type of the server. Possible values are: `local`, `bootscript` or `rescue`.
 
-- `bootscript` - The ID of the bootscript to use
+- `bootscript` - The ID of the bootscript to use  (set boot_type to `bootscript`).
```
Comment (kindermoumoute) [inline docs/resources/instance_server.md:180]: bootscript_id

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

**Review by Sh4d1 — CHANGES_REQUESTED** [review]
Need to set the bootype in create req event when no bootscript

Some questions:
- what about the update?
- what if the user sets the boot_type to local and a bootscript ID ? The states will mismatch, I think throwing an error would be better

**Comment by NoSuchCommand** [comment]
> Need to set the bootype in create req event when no bootscript
> 
> Some questions:
> 
>     * what about the update?
> 
>     * what if the user sets the boot_type to local and a bootscript ID ? The states will mismatch, I think throwing an error would be better

File: scaleway/resource_instance_server.go [hunk 5]
```diff
@@ -259,6 +271,12 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 		Tags:              expandStrings(d.Get("tags")),
 	}
 
+	if bootScriptID, ok := d.GetOk("bootscript_id"); ok {
+		b := instance.BootTypeBootscript
```
Comment (kindermoumoute) [inline scaleway/resource_instance_server.go:275]: IIRC a bootscript can be run in rescue mode too, that's why you should read `boot_type` attribute too

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: scaleway/resource_instance_server.go [hunk 6]
```diff
@@ -272,11 +272,16 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 	}
 
 	if bootScriptID, ok := d.GetOk("bootscript_id"); ok {
-		b := instance.BootTypeBootscript
-		req.BootType = &b
 		req.Bootscript = expandStringPtr(bootScriptID)
 	}
 
+	if bootType, ok := d.GetOk("boot_type"); ok {
+		bootType, ok := bootType.(instance.BootType)
```
Comment (kindermoumoute) [inline scaleway/resource_instance_server.go:279]: I think you mean `bootType := instance.BootType(bootType.(string))`

Comment (kindermoumoute) [reply scaleway/resource_instance_server.go:279]: https://play.golang.org/p/80ZOT8_UdkG

**Review by kindermoumoute — DISMISSED** [review]
LGTM

File: scaleway/resource_instance_server.go [hunk 7]
```diff
@@ -259,6 +271,15 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
 		Tags:              expandStrings(d.Get("tags")),
 	}
 
+	if bootScriptID, ok := d.GetOk("bootscript_id"); ok {
+		req.Bootscript = expandStringPtr(bootScriptID)
+	}
+
+	if bootType, ok := d.GetOk("boot_type"); ok {
+		bootType := instance.BootType(bootType.(string))
```
Comment (Sh4d1) [inline scaleway/resource_instance_server.go:279]: remove this line

**Review by Sh4d1 — CHANGES_REQUESTED** [review]

**Review by Sh4d1 — APPROVED** [review]
LGTM

**Pull request merged**